opensync-ext-cloud/src/main/java/com/telecominfraproject/wlan/opensync/external/integration/utils/MqttStatsPublisher.java

```diff
@@ -208,8 +208,11 @@ public void processMqttMessage(String topic, Report report) {
 
             if (!metricRecordList.isEmpty()) {
                 long serviceMetricTimestamp = System.currentTimeMillis();
-                metricRecordList.stream().forEach(smr -> {              
-                    smr.setCreatedTimestamp(serviceMetricTimestamp);
+                metricRecordList.stream().forEach(smr -> {  
```

Comment:
Why not just move the System.currentTimeMillis() inside the forEach, then it will take a different value for each one.

Reply:
That could work too

Reply:
It's not guaranteed to get a different time , which is why we'd let the CloudEventDispatcherController generate the time by incrementing each by 1 msec. This is the same problem/solution that Lynn had fixed for generating 2 alarms at the same time. It's just a beautiful design.

Reply:
This is ridiculous. Why not just have the CloudEventDispatcherController generate the time, and ignore whatever timestamps are being passed it's way? 
The logic here is broken. The metrics in the record already have their timestamps perverted from what they originally were, to a System.currentTimeMillis, because there are problems between he AP times and the cloud times. Maybe just go back to setting '0' or just don't set anything at all? Accuracy is out the window.
Also, why use timestamp for the unique key?

Reply:
Sorry, just looked at it. The problem is not even what the CloudEventDispatcher is doing, it is that somewhere down the line a decision has been made to treat timestamps as being unique keys!
This is a very flawed assumption, not all received timestamps are even guaranteed to be in milliseconds precision

Reply:
That is correct, the Metrics Cassandra database schema is flawed... and I'm being polite saying that!